theano/tensor/nnet/tests/test_sigm.py

```diff
@@ -430,7 +421,8 @@ def test_log1pexp_to_softplus(self):
 
         out = T.log(1 + T.exp(x))
         f = theano.function([x], out, mode=self.m)
-        assert hasattr(f.maker.fgraph.outputs[0].tag, 'trace')
+
+        assert check_stack_trace(f, ops_to_check='all')
         topo = f.maker.fgraph.toposort()
         assert len(topo) == 1
         assert isinstance(topo[0].op.scalar_op,
```

Comment:
Actually, the optimizations inserting softplus are done through PatternSub (see https://github.com/lamblin/Theano/blob/fix_stack_check_nnet/theano/tensor/nnet/sigm.py#L406), which has no way of preserving stack traces yet.